### PR TITLE
Add support for pagination parameters to the projects endpoint (2/3)

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
- added pagination options to list_project_files & list_projects
- fixed erroneous type hints
- fixed failing pre-commit config

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-2873 Add pagination to the QFC API endpoints](https://app.clickup.com/t/2192114/QF-2873)

File: src/qfieldcloud_sdk/sdk.py [hunk 1]
```diff
@@ -117,21 +118,35 @@ def logout(self) -> None:
...
+        self, project_id: str, limit: Optional[int] = None, offset: Optional[int] = None
+    ) -> List[Dict[str, Any]]:
+        params = {}
+
+        if limit:
+            params["limit"] = limit
+        if offset:
+            params["offset"] = offset
+
+        resp = self._request("GET", f"files/{project_id}", params=params)
```
Comment (suricactus) [inline src/qfieldcloud_sdk/sdk.py:149]: But this endpoint is not using pagiation, no?

File: src/qfieldcloud_sdk/sdk.py [hunk 2]
```diff
@@ -117,21 +118,35 @@ def logout(self) -> None:
...
-            },
-        )
+        self,
+        username: Optional[str] = None,
+        include_public: Optional[bool] = False,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> List[Dict[str, Any]]:
+        """Returns a paginated lists the project of a given user. If the user is omitted, it fallbacks to the currently logged in user"""
+        params = {"include-public": "1" if include_public else "0"}
```
Comment (suricactus) [inline src/qfieldcloud_sdk/sdk.py:128]:
```suggestion
        params = {
            "include-public": int(include_public),
        }
```

I think it is safe to convert the `Optional[bool]` to `int` like this. No idea why initially didn't write it this way.

File: src/qfieldcloud_sdk/cli.py [hunk 3]
```diff
@@ -186,14 +200,28 @@ def list_projects(ctx, include_public):
...
 @click.argument("project_id")
 @click.pass_context
-def list_files(ctx, project_id):
+def list_files(ctx, project_id, **opts):
     """List QFieldCloud project files."""
 
     log(f'Getting file list for "{project_id}"…')
 
-    files = ctx.obj["client"].list_remote_files(project_id)
+    files = ctx.obj["client"].list_remote_files(project_id, **opts)
```
Comment (suricactus) [inline src/qfieldcloud_sdk/cli.py:224]: But this endpoint is not using pagiation, no?

**Review by suricactus — COMMENTED** [review]
I would give green light for some history rewriting. Don't squash things that are completely unrelated into one commit, e.g. the CI changes and the code.

File: src/qfieldcloud_sdk/sdk.py (hunk 2 above)
Comment (why-not-try-calmer) [reply src/qfieldcloud_sdk/sdk.py:128]:
```python
int(False) # ok
int(True) # ok
int(None) # TypeError
```

**Review by suricactus — APPROVED** [review]

**Pull request merged**